packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -847,17 +859,19 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
                                 context,
                                 widget.route.animation!,
                                 widget.route.secondaryAnimation!,
-                                // This additional AnimatedBuilder is include because if the
-                                // value of the userGestureInProgressNotifier changes, it's
-                                // only necessary to rebuild the IgnorePointer widget and set
-                                // the focus node's ability to focus.
+                                // _listenable updates when this route's animations change
+                                // values, but the _ignorePointerNotifier can also update
+                                // when the status of animations on popping routes change,
+                                // even when this route's animations' values don't. Also,
+                                // when the value of the _ignorePointerNotifier changes,
+                                // it's only necessary to rebuild the IgnorePointer
+                                // widget and set the focus node's ability to focus.
                                 AnimatedBuilder(
-                                  animation: widget.route.navigator?.userGestureInProgressNotifier ?? ValueNotifier<bool>(false),
+                                  animation: widget.route._ignorePointerNotifier,
                                   builder: (BuildContext context, Widget? child) {
-                                    final bool ignoreEvents = _shouldIgnoreFocusRequest;
-                                    focusScopeNode.canRequestFocus = !ignoreEvents;
+                                    focusScopeNode.canRequestFocus = !_shouldIgnoreFocusRequest;
                                     return IgnorePointer(
-                                      ignoring: ignoreEvents,
+                                      ignoring: widget.route._ignorePointer,
```

Comment:
what about _shouldIgnoreFocusRequest?

Reply:
Ah yes thank you for bringing this up, I had this question as well and forgot to comment on it—the code before was written such that it's following the same logic, but I couldn't quite tell if that was incidental to the fact that they happened to share the same conditions at the time.

It sounds like you'd suggest that we have the focusNodeScope should also be set to the value of the notifier? For context, the cases where `_ignorePointer` is true is currently a super set of the cases where `_shouldIgnoreFocusRequest` is true

Reply:
> For context, the cases where _ignorePointer is true is currently a super set of the cases where _shouldIgnoreFocusRequest is true

Is this true? I am not sure if it is already fixed, but if I remember correctly, the cupertino back swipe can swipe the animation to completed/dismissed if you swipe to the edges but still holding your finger...

Reply:
I do also get some failing tests when I do this:
- `focus traverse correct when pop multiple page simultaneously`
- `focus traversal is correct when popping multiple pages simultaneously - with focused children`

Thoughts? I'm a bit unfamiliar with how the focus stuff works but happy to look into it further

Reply:
Yup you're correct: when you start a gesture, and then take it back to the start, if you hold your finger down then the gesture will be in progress but my current logic wouldn't ignore pointers there. I'll make an update to that, because I think it probably should continue to block interactions